kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -86,13 +97,48 @@ private TableConfig(
      * @return the value of the table property
      */
     public T fromMetadata(Metadata metadata) {
-        T value = fromString.apply(metadata.getConfiguration().getOrDefault(key, defaultValue));
-        if (!validator.test(value)) {
-            throw new IllegalArgumentException(
-                    String.format("Invalid value for table property '%s': '%s'. %s",
-                            key, value, helpMessage));
+        String value = metadata.getConfiguration().getOrDefault(key, defaultValue);
+        validate(value);
+        return fromString.apply(value);
+    }
+
+    /**
+     * Returns the key of the table property.
+     *
+     * @return the key of the table property
+     */
+    public String getKey() {
+        return key;
+    }
+
+    /**
+     * Validates the given properties.
+     *
+     * @param configurations the properties to validate
+     * @throws IllegalPropertyValueException if any of the properties are invalid
+     * @throws UnknownConfigurationKeyException if any of the properties are unknown
+     */
+    public static void validateProperties(Map<String, String> configurations) {
+        for (Map.Entry<String, String> kv : configurations.entrySet()) {
+            String key = kv.getKey().toLowerCase(Locale.ROOT);
+            String value = kv.getValue();
+            if (key.startsWith("delta.")) {
+                TableConfig tableConfig = entries.get(key);
+                if (tableConfig != null) {
+                    tableConfig.validate(value);
+                } else {
+                    throw DeltaErrors.unknownConfigurationKeyException(key);
+                }
+            } else {
+                throw DeltaErrors.unknownConfigurationKeyException(key);
+            }
         }
-        return value;
     }
-}
 
+    private void validate(String value) {
+        T parsedValue = fromString.apply(value);
+        if (!validator.test(parsedValue)) {
+            throw DeltaErrors.illegalPropertyValueException(key, value, helpMessage);
```

Comment:
We should try to keep the terminology consistent here: `unknownConfigurationKeyException` and `illegalPropertyValueException`. It should be either Configuration or Property.